### PR TITLE
Adds Array <-> List transforms

### PR DESCRIPTION
**Pull request by rstegg**
Shorted out previous #63 PR, here are review changes:

- [x] Remove README examples
- [x] Fix arrayToList signatures
- [x] Fix listToArray signatures
- [x] Fix maybeToEither signatures
- [x] Fix arrayToList specs
- [x] Fix listToArray specs

This pr will result in the following task list of issue #61:

- [ ] Remove Array constraint on List constructor, and have it work like of breaking
- [x] Add fromArray to List constructor [PR #62]
- [x] Add toArray to a List instance [PR #62]
- [x] Add listToArray transformation function.
- [x] Add arrayToList transformation function.

### CODE REVIEW

File: transforms/arrayToList.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/** @license ISC License (c) copyright 2017 original and current authors */
+/** @author Ian Hofmann-Hicks (evil) */
+
+const isFunction = require('../predicates/isFunction')
+const isArray = require('../predicates/isArray')
```
Comment (evilsoft) [inline transforms/arrayToList.js]: Alphabetize like functions.

File: transforms/arrayToList.spec.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+const test = require('tape')
+const helpers = require('../test/helpers')
+
+const bindFunc = helpers.bindFunc
+
+const identity = require('../combinators/identity')
+
+const isFunction = require('../predicates/isFunction')
```
Comment (evilsoft) [inline transforms/arrayToList.spec.js]: Alphabetize like functions.

File: transforms/arrayToList.spec.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  const a = arrayToList(data)
+
+  t.ok(isSameType(List, a), 'returns a List from an Array')
+  t.same(a.value(), data, 'returns expected List')
+
+  t.end()
+})
+
+test('arrayToList with List returning function', t => {
```
Comment (evilsoft) [inline transforms/arrayToList.spec.js]: 👮-y paste issue here.
should be 'arrayToList with Array returning function'

File: transforms/arrayToList.spec.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  t.throws(f(false), TypeError, 'throws if function returns false')
+  t.throws(f(true), TypeError, 'throws if function returns true')
+  t.throws(f({}), TypeError, 'throws if function returns an object')
+
+  const data = [32, 54, 'b', [24, 65], ['91']]
+
+  const m = arrayToList(identity, data)
+
+  t.ok(isSameType(List, m), 'returns a List from a function returning an Array')
+  t.same(m.value(), data, 'provides List of Array')
```
Comment (evilsoft) [inline transforms/arrayToList.spec.js]: The test should probably read something like:

'does not modify the structure of underlying data'

File: transforms/listToArray.spec.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
+const test = require('tape')
+const helpers = require('../test/helpers')
+
+const bindFunc = helpers.bindFunc
+
+const identity = require('../combinators/identity')
+
+const isFunction = require('../predicates/isFunction')
```
Comment (evilsoft) [inline transforms/listToArray.spec.js]: Alphabetize like functions.

File: transforms/listToArray.spec.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  t.throws(f({}), TypeError, 'throws if function returns an object')
+  t.throws(f([]), TypeError, 'throws if function returns an array')
+
+  const data = [32, 54, 'b', [24, 65], ['91']]
+  const list = List.fromArray(data)
+
+  const m = listToArray(identity, list)
+
+  t.ok(isArray(m), 'returns an Array from a function returning a List')
+  t.same(m, data, 'provides wrapped array from list')
```
Comment (evilsoft) [inline transforms/listToArray.spec.js]: The test should probably read something like:

'does not modify the structure of underlying data'

File: transforms/arrayToList.spec.js (hunk 3 above)
Comment (rstegg) [inline transforms/arrayToList.spec.js]: oh no, they found me! 🏃

File: transforms/listToArray.spec.js (hunk 6 above)
Comment (rstegg) [inline transforms/listToArray.spec.js]: 'preserves the structure of underlying data' conforms to category theory

Comment (evilsoft) [inline transforms/listToArray.spec.js]: Sure. That is something like that.

**Pull request merged**